service/storage_service.cc

```diff
@@ -365,6 +370,21 @@ void storage_service::prepare_to_join(std::vector<inet_address> loaded_endpoints
         }
     }
 
+    if (replacing_a_node_with_same_ip) {
+        slogger.info("Replacing a node with same IP address, my address={}, node being replaced={}",
+                    get_broadcast_address(), get_broadcast_address());
+        slogger.info("Update tokens for replacing node early, replacing node has the same IP address of the node being replaced");
+        _token_metadata.update_normal_tokens(_bootstrap_tokens, get_broadcast_address());
+    }
+
+    if (replacing_a_node_with_diff_ip) {
+        // replacing_a_node_with_diff_ip guarantees replace_address contains a value
+        auto replace_address = db().local().get_replace_address();
```

Comment:
Suggestion: `on_internal_error` in case there's a bug and we failed to satisfy the guarantee (so `!replace_address`)

Reply:
I used to have a check on if replace_address is engaged to be defensive. Tomek suggested not to check replace_address here again, because we have

replacing_a_node_with_diff_ip = replace_address && *replace_address != get_broadcast_address();

replacing_a_node_with_diff_ip will be set true only when replace_address is engaged.

Reply:
Ok.